packages/core/src/document.ts

```diff
@@ -580,39 +581,73 @@ class Document extends EventEmitter implements DocStateHolder {
     await this.dispatcher.publishTip(this.id, this.tip)
   }
 
+  /**
+   * Returns whether the given CID is in the log, and there is a later log entry that is an anchor commit
+   * @param cid
+   */
+  async _isCidAnchored(cid: CID): Promise<boolean> {
+    let foundCid = false
+    for (const entry of this.state.log) {
+      if (entry.cid.equals(cid)) {
+        foundCid = true
+      }
+
+      if (foundCid && entry.type == CommitType.ANCHOR) {
+        return true
+      }
+    }
+    return false
+  }
+
   /**
    * Request anchor for the latest document state
    */
   async anchor (): Promise<void> {
-    this._context.anchorService.on(this.id.toString(), async (asr: AnchorServiceResponse): Promise<void> => {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const doc = this
+    const tip: CID = this.tip
+
+    this._context.anchorService.on(this.id.toString(), async function listener (asr: AnchorServiceResponse) {
+      if (!asr.cid.equals(tip)) {
+        // This message is about a different tip for the same document
+        return
+      }
```

Comment:
Why do we need the condition if `FAILED` if we just return here when the cid is not equal to the tip?

Reply:
I don't understand the question.  What condition are you referring to?

Reply:
Sorry that was way to unclear!